system/lib/compiler-rt/emscripten_setjmp.c

```diff
@@ -73,33 +74,54 @@ void emscripten_longjmp(uintptr_t env, int val) {
 #endif
 
 #ifdef __USING_WASM_SJLJ__
-
 struct __WasmLongjmpArgs {
   void *env;
   int val;
 };
+#endif
 
-thread_local struct __WasmLongjmpArgs __wasm_longjmp_args;
+// jmp_buf should have large enough size and alignment to contain
+// this structure.
+struct jmp_buf_impl {
+  void* func_invocation_id;
+  uint32_t label;
+#ifdef __USING_WASM_SJLJ__
+  struct __WasmLongjmpArgs arg;
+#endif
+};
 
-// llvm uses `1` for the __c_longjmp tag.
-// See https://github.com/llvm/llvm-project/blob/main/llvm/include/llvm/CodeGen/WasmEHFuncInfo.h
-#define C_LONGJMP 1
+void __wasm_setjmp(void* env, uint32_t label, void* func_invocation_id) {
+  struct jmp_buf_impl* jb = env;
+  assert(label != 0);                 // ABI contract
+  assert(func_invocation_id != NULL); // sanity check
+  jb->func_invocation_id = func_invocation_id;
+  jb->label = label;
+}
 
-// Wasm EH allows us to throw and catch multiple values, but that requires
-// multivalue support in the toolchain, whch is not reliable at the time.
-// TODO Consider switching to throwing two values at the same time later.
-void __wasm_longjmp(void *env, int val) {
-  __wasm_longjmp_args.env = env;
+uint32_t __wasm_setjmp_test(void* env, void* func_invocation_id) {
+  struct jmp_buf_impl* jb = env;
+  assert(jb->label != 0);             // ABI contract
+  assert(func_invocation_id != NULL); // sanity check
+  if (jb->func_invocation_id == func_invocation_id) {
+    return jb->label;
+  }
+  return 0;
+}
+
+#ifdef __USING_WASM_SJLJ__
+void __wasm_longjmp(void* env, int val) {
+  struct jmp_buf_impl* jb = env;
+  struct __WasmLongjmpArgs* arg = &jb->arg;
   /*
-￼  * C standard:
-￼  *   The longjmp function cannot cause the setjmp macro to return
-￼  *   the value 0; if val is 0, the setjmp macro returns the value 1.
-￼  */
+   * C standard says:
+   * The longjmp function cannot cause the setjmp macro to return
+   * the value 0; if val is 0, the setjmp macro returns the value 1.
+   */
```

Comment:
I know it's preexisting comment, but how about changing it to `//` style comment so that it's consistent with other comments?